src/platform/Linux/ThreadStackManagerImpl.h

```diff
@@ -93,25 +89,26 @@ class ThreadStackManagerImpl : public ThreadStackManager
     static ThreadStackManagerImpl sInstance;
 
 private:
-    struct DBusConnectionDeleter
-    {
-        void operator()(DBusConnection * aConnection)
-        {
-            dbus_connection_close(aConnection);
-            dbus_connection_unref(aConnection);
-        }
-    };
+    static constexpr char kDBusOpenThreadService[]    = "io.openthread.BorderRouter.wpan0";
+    static constexpr char kDBusOpenThreadObjectPath[] = "/io/openthread/BorderRouter/wpan0";
+
+    static constexpr char kOpenthreadDeviceRoleDisabled[] = "disabled";
+    static constexpr char kOpenthreadDeviceRoleDetached[] = "detached";
+    static constexpr char kOpenthreadDeviceRoleChild[]    = "child";
+    static constexpr char kOpenthreadDeviceRoleRouter[]   = "router";
+    static constexpr char kOpenthreadDeviceRoleLeader[]   = "leader";
+
+    static constexpr char kPropertyDeviceRole[] = "DeviceRole";
 
-    using UniqueDBusConnection = std::unique_ptr<DBusConnection, DBusConnectionDeleter>;
+    std::unique_ptr<OpenthreadIoOpenthreadBorderRouter, decltype(&g_object_unref)> mProxy;
```

Comment:
Could we define deleters like DBusConnectionDeleter and named types like UniqueDBusConnection to avoid the decltype here?

Reply:
I'd like to argue that `decltype(&g_object_unref)` is more cleaner than creating a deleter. The only pitfall is that it stores a pointer rather than a empty object. But without empty member optimization (using `[[no_unique_address]]` of C++20), it is still requiring a word to store the deleter.

So I'm unclear why `decltype` is bad here.